### PR TITLE
Initial commit of Lamp Stack in IBM Cloud gen 2

### PR DESCRIPTION
**Pull request by jjalvare**

### CODE REVIEW

File: terraform/lamp-stack-ibmcloud/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Deploy a LAMP stack using Terraform on IBM Cloud Gen 2
```
Comment (jwcroppe) [inline terraform/lamp-stack-ibmcloud/README.md]: s/IBM Cloud Gen 2/IBM Cloud VPC throughout

File: terraform/lamp-stack-ibmcloud/README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Deploy a LAMP stack using Terraform on IBM Cloud Gen 2 
+
+Deploy a Linux Apache Mariadb PHP (LAMP) stack on Ubuntu 18.04 ppc64le running in IBM Cloud Gen 2
+
+## Pre-Requisites
```
Comment (jwcroppe) [inline terraform/lamp-stack-ibmcloud/README.md]: s/Pre-Requisites/Prerequisites

File: terraform/lamp-stack-ibmcloud/README.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+## Pre-Requisites 
+
+1. Terraform >= 0.11.14 
+2. IBM Cloud Terraform Provider >= 0.20.0 
+3. IBM Cloud API Key 
+
+## Overview
+
+This deployment provisions a LAMP Stack running in IBM Cloud on a  Ubuntu ppc64le VM.
```
Comment (jwcroppe) [inline terraform/lamp-stack-ibmcloud/README.md]: s/Stack/stack
an Ubuntu (single place)

File: terraform/lamp-stack-ibmcloud/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+## Overview
+
+This deployment provisions a LAMP Stack running in IBM Cloud on a  Ubuntu ppc64le VM. 
+
+After install you can access the default web page of Ubuntu VM by IP:
+
+    https://<IP>/  
+
+To validate Mariadb successful connection:
```
Comment (jwcroppe) [inline terraform/lamp-stack-ibmcloud/README.md]: MariaDB

File: terraform/lamp-stack-ibmcloud/main.tf [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+  public_key = "${tls_private_key.ssh_key_keypair.public_key_openssh}"
+}
+
+# Create a public floating IP so that the app is available on the Internet
+resource "ibm_is_floating_ip" "fip1" {
+  name = "${var.basename}-fip"
+  target = "${ibm_is_instance.vm.primary_network_interface.0.id}"
+}
+
+# Enable ssh into the instance for debug
```
Comment (jwcroppe) [inline terraform/lamp-stack-ibmcloud/main.tf:37]: Should we disable by default and make an option to enable via a property?

File: terraform/lamp-stack-ibmcloud/scripts/setup_env.sh [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+# wait for setup
+while ! grep "Cloud-init .* finished" /var/log/cloud-init.log; do
+    echo "$(date -Ins) Waiting for cloud-init to finish"
+    sleep 2
+done
+until sudo apt-get update; do sleep 5; done # some hacky way to get apt-get to succeed at boot 
+# install apache mariadb php 
+sudo apt-get install software-properties-common -y
+sudo apt-key adv --recv-keys --keyserver hkp://keyserver.ubuntu.com:80 0xF1656F24C74CD1D8
+sudo add-apt-repository 'deb [arch=amd64,arm64,ppc64el] http://mirror.i3d.net/pub/mariadb/repo/10.4/ubuntu bionic main'
```
Comment (jwcroppe) [inline terraform/lamp-stack-ibmcloud/scripts/setup_env.sh]: Is ppc64le not enabled by default in NG Cloud?

File: terraform/lamp-stack-ibmcloud/scripts/setup_env.sh [hunk 7]
```diff
@@ -0,0 +1,49 @@
+#!/bin/bash -xe
```
Comment (jwcroppe) [inline terraform/lamp-stack-ibmcloud/scripts/setup_env.sh]: Can we get a high level header about what this file does?

File: terraform/lamp-stack-ibmcloud/variables.tf [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Licensed Materials - Property of IBM
+#
+# ©Copyright IBM Corp. 2017
```
Comment (jwcroppe) [inline terraform/lamp-stack-ibmcloud/variables.tf]: Copyrights should all be 2020

**Review by jwcroppe — CHANGES_REQUESTED** [review]
Nice job - a few nit changes and then we can merge.

File: terraform/lamp-stack-ibmcloud/scripts/setup_env.sh (hunk 6 above)
Comment (jjalvare) [reply terraform/lamp-stack-ibmcloud/scripts/setup_env.sh]: Correct. I don't need to add repo.

File: terraform/lamp-stack-ibmcloud/scripts/setup_env.sh (hunk 7 above)
Comment (jjalvare) [reply terraform/lamp-stack-ibmcloud/scripts/setup_env.sh]: Will do.

File: terraform/lamp-stack-ibmcloud/README.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+3. IBM Cloud API Key with authorization to provision in IBM Cloud VPC and IBM Cloud Block Storage for VPC 
+
+## Overview
+
+This deployment provisions:
+  1. a Linux Ubuntu ppc64le VM server on IBM Cloud VPC 
+  3. Creates a 200GB Block Storage for VPC  
+  4. Creates a new ssh key to login 
+  5. Opens Ports 80, 443 in VPC Security Group port to access web page 
+  6. Opens Ports 22 to access SSH console
```
Comment (jwcroppe) [inline terraform/lamp-stack-ibmcloud/README.md:18]: s/Ports/port

File: terraform/lamp-stack-ibmcloud/README.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+
+## Prerequisites 
+
+1. Terraform >= 0.11.14 
+2. IBM Cloud Terraform Provider >= 0.20.0 
+3. IBM Cloud API Key with authorization to provision in IBM Cloud VPC and IBM Cloud Block Storage for VPC 
+
+## Overview
+
+This deployment provisions:
```
Comment (jwcroppe) [inline terraform/lamp-stack-ibmcloud/README.md:13]: s/ provisions//

File: terraform/lamp-stack-ibmcloud/README.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+  1. a Linux Ubuntu ppc64le VM server on IBM Cloud VPC
```
Comment (jwcroppe) [inline terraform/lamp-stack-ibmcloud/README.md:14]: s/a/Create a

**Pull request merged**